hal/halx86/amd64/x86bios.c

```diff
@@ -44,71 +46,104 @@ DbgDumpPage(PUCHAR MemBuffer, USHORT Segment)
 VOID
 NTAPI
 HalInitializeBios(
-    _In_ ULONG Unknown,
+    _In_ ULONG Phase,
     _In_ PLOADER_PARAMETER_BLOCK LoaderBlock)
 {
     PPFN_NUMBER PfnArray;
     PFN_NUMBER Pfn, Last;
     PMEMORY_ALLOCATION_DESCRIPTOR Descriptor;
     PLIST_ENTRY ListEntry;
     PMDL Mdl;
+    ULONG64 PhysicalAddress;
 
-    /* Allocate an MDL for 1MB */
-    Mdl = IoAllocateMdl(NULL, 0x100000, FALSE, FALSE, NULL);
-    if (!Mdl)
+    if (Phase == 0)
     {
-        ASSERT(FALSE);
-    }
+        PhysicalAddress = HalpAllocPhysicalMemory(LoaderBlock,
+                                                  0x100000,
```

Comment:
Is there some #define for this quantity?

Reply:
no